src/domain/community/virtual-contributor/virtual.contributor.service.ts

```diff
@@ -546,4 +551,17 @@ export class VirtualContributorService {
 
     return virtualContributorMatchesCount;
   }
+
+  //adding this to avoid circular dependency between VirtualContributor, Room, and Invitation
+  private async deleteVCInvitations(contributorID: string) {
+    const invitations = await this.entityManager.find(Invitation, {
+      relations: { community: true },
+      where: { invitedContributor: contributorID },
+    });
+    for (const invitation of invitations) {
+      if (invitation.authorization)
+        await this.authorizationPolicyService.delete(invitation.authorization);
+      await this.entityManager.remove<Invitation>(invitation);
```

Comment:
i don't think you need `Invitation` here, furthermore the type should be infered by the type of `invitation`